Sources/Citadel/Server.swift

```diff
@@ -13,42 +10,128 @@ final class CloseErrorHandler: ChannelInboundHandler {
-//                    NIOSSHHandler(
-//                        role: .server(
-//                            .init(
-//                                hostKeys: [hostKey],
-//                                userAuthDelegate: serverDelegate,
-//                                globalRequestDelegate: serverDelegate
-//                            )
-//                        ),
-//                        allocator: channel.allocator,
-//                        inboundChildChannelInitializer: serverDelegate.initializeSshChildChannel
-//                    ),
-//                    CloseErrorHandler()
-//                ])
-//            }
-//            .serverChannelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR), value: 1)
-//            .serverChannelOption(ChannelOptions.socket(SocketOptionLevel(IPPROTO_TCP), TCP_NODELAY), value: 1)
-//
-//        return bootstrap.bind(host: "0.0.0.0", port: 2222).map { channel in
-//            SSHServer(channel: channel)
-//        }
-//    }
-//}
+final class SubsystemHandler: ChannelDuplexHandler {
+    typealias InboundIn = SSHChannelData
+    typealias InboundOut = SSHChannelData
+    typealias OutboundIn = SSHChannelData
+    typealias OutboundOut = SSHChannelData
+    
+    let sftp: SFTPDelegate?
+    
+    init(sftp: SFTPDelegate?) {
+        self.sftp = sftp
+    }
+    
+    func handlerAdded(context: ChannelHandlerContext) {
+        context.channel.setOption(ChannelOptions.allowRemoteHalfClosure, value: true).whenFailure { error in
+            context.fireErrorCaught(error)
+        }
+    }
+    
+    func channelInactive(context: ChannelHandlerContext) {
+        context.fireChannelInactive()
+    }
+    
+    func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
+        switch event {
+        case let event as SSHChannelRequestEvent.SubsystemRequest:
+            switch event.subsystem {
+            case "sftp":
+                guard let sftp = sftp else {
+                    context.channel.close(promise: nil)
+                    return
+                }
+                
+                _ = SFTPServerSubsystem.setupChannelHanders(
+                    channel: context.channel,
+                    delegate: sftp,
+                    logger: .init(label: "nl.orlandos.citadel.sftp-server")
+                )
+            default:
+                context.fireUserInboundEventTriggered(event)
+            }
+        case ChannelEvent.inputClosed:
+            context.channel.close(promise: nil)
+        default:
+            context.fireUserInboundEventTriggered(event)
+        }
+    }
+    
+    func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+        context.fireChannelRead(data)
+    }
+    
+    func write(context: ChannelHandlerContext, data: NIOAny, promise: EventLoopPromise<Void>?) {
+        context.write(data, promise: promise)
+    }
+}
+
+final class CitadelServerDelegate {
```

Comment:
Is this a new and non protocol version of the removed `SSHServerDelegate` ? Won't it need those same protocol conformances that used to have or is the similarity just coincidence?